examples/systemcalls/systemcalls.c

```diff
@@ -92,8 +173,73 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    /* taken directly from stackoverflow reference */
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0664);
+    if(fd == -1)
+    {
+        perror("failed to open file");
+        return false;
+    }
+
+    /* var to store process status from wait() */
+    int wstat   = 0;
+    /* store return from fork and wait */
+    pid_t pid   = 0;
+    pid_t wpid  = 0;
+
+    /* flush stdout for readability */
+    fflush(stdout);
+
+    /* create new child process */
+    pid = fork();
+    /* error handling for fork, returns -1 if child can't be created */
+    if(pid == -1)
+    {
+        perror("fork() failed to create child process");
+        return false;
+    }
+    else if(pid == 0)
+    {
+        /* redirect stdout to opened file */
+        if(dup2(fd, 1) < 0)
+        {
+            perror("dup2 failed to redirect output");
+            return false;
+        }
+        close(fd);
+        /* change newly created process with given commands */
+        execv(command[0], command);
+        perror("execv failed to alter child process");
+        exit(-1);
+    }
+    /* Parent process, fork returns pid of created child */
+    else
+    {
+        close(fd);
+        /* wait for child to exit */
+        wpid = waitpid(pid, &wstat, 0);
+        if(wpid == -1)
+        {
+            perror("waitpid failure");
+            return false;
+        }
+        /* check exit status of child, WIFEXITED indicates successful termination */
+        if(WIFEXITED(wstat) == true)
+        {
+            if(wstat)
+                return false;
```

Comment:
Missing error handling in these return paths. 
Also see discussion in class about avoiding duplication of logic on these two very similar functions.